submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/ext/ExtendedPageMapper.kt

```diff
@@ -0,0 +1,30 @@
+package ac.uk.ebi.biostd.submission.web.resources.ext
+
+import ac.uk.ebi.biostd.submission.web.model.ExtPage
+import ebi.ac.uk.extended.model.ExtSubmission
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import java.net.URI
+
+class ExtendedPageMapper(private val instanceBase: URI) {
+
+    fun asExtPage(page: Page<ExtSubmission>): ExtPage =
+            ExtPage(page.content, page.totalPages, getNextPage(page), getPreviousPage(page))
+
+    private fun getPreviousPage(page: Page<ExtSubmission>): String? {
```

Comment:
These could be one liners. Also, I'm not sure about this syntax. Maybe the if/else would be clearer in these cases. Anyway, just a suggestion.

I miss you, ternary operator!! 😢